### DISCUSSION TITLE
Switch to pyproject.toml-based package

### DISCUSSION THREAD

**Discussion by schuenke** [imr-framework/pypulseq#124] [General]
[PEP 518 – Specifying Minimum Build System Requirements for Python Projects](https://peps.python.org/pep-0518/) suggests to use a `pyproject.toml` file to specify the build dependencies and provide all project-related metadata and Python.org adjusted their [Packaging Tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/) accordingly. 

PyPulseq currently still uses the "old" concept of a `setup.py` file, but the ongoing refactoring process (#95, #123) might be the right moment to switch to a ``pyproject.toml` file.

The advantage of a `pyproject.toml` file is that it can also be used as config file for almost all current linters and formatters. A simple example of a `pyproject.toml` file including some additional tool configurations can be found in the [pyproject-toml-cheat-sheet Github Repo](https://github.com/ricardolsmendes/pyproject-toml-cheat-sheet/blob/main/pyproject.toml).

What are your opinions?

**Comment by sravan953** [comment]
Let's do it. I think this should be in a dedicated branch before merging into dev, and then main.

**Comment by schuenke** [comment]
Changed with #195 already
